### PR TITLE
Indirect - Corrections - Ensure Plot Current Preview plots transformed container in Container Subtraction and Apply Paalman Pings

### LINKED ISSUE #21982: ISIS Indirect Unscripted Testing :: Corrections > Container Subtraction and Apply Abs Corr
* Plot Current Preview button plots the current workspace indexes from the loaded data but does not apply the shift and scale. This should plot the same as the preview plot, or the the button name should be changed to a better description of what it does.
  * Additionally, plot current preview does not plot the subtracted data either

*This is true of both the `Container Subtraction` and `Apply Absorption corrections` interfaces*

### PR DESCRIPTION
**Pull request by brandonhewer**
In ```Interfaces > Indirect > Corrections```, the ```Container Subtraction``` and ```Apply Paalman Pings``` interfaces do not plot the transformed container or the result when clicking ```Plot Current Preview```.

This PR ensures the transformed container and the result are plot, when appropriate, by the ```Plot Current Preview``` feature.

**To test:**
Container Subtraction
1. Navigate to the ```Interfaces > Indirect > Corrections > Container Subtraction``` interface.
2. In the ```Sample``` field, provide the supplied irs26176* file.
3. In the ```Container``` field, provide the supplied irs26173* file.
4. Check the check-box labelled ```Scale Container by factor``` and enter 0.9 into the associated spinner.
5. Check the check-box labelled ```Shift x-values of container by adding``` and enter 0.1 into the associated spinner.
6. Click 'Run'.
7. Once the container subtraction is complete, click ```Plot Current Preview```.
8. Ensure the sample, container and result of the subtraction are plot in both the preview within the interface and the external plot.

Apply Absorption Corrections
1. Navigate to the ```Interfaces > Indirect > Corrections > Apply Absorption Corrections``` interface.
2. In the ```Sample``` field, provide the supplied irs26176* file.
3. In the ```Corrections``` field, provide the supplied *_Corrections file.
4. Check the check-box labelled ```Use Container``` and provide the supplied irs26173* file.
5. Check the check-box labelled ```Scale Container by factor``` and enter 0.9 into the associated spinner.
6. Check the check-box labelled ```Shift x-values of container by adding``` and enter 0.1 into the associated spinner.
7. Click 'Run'.
8. Once the algorithm is complete, click ```Plot Current Preview```.
9. Ensure the sample, container and result of applying the corrections are plot in both the preview within the interface and the external plot.

**Test Files:**
[corrections-test-files.zip](https://github.com/mantidproject/mantid/files/1796473/corrections-test-files.zip)

Fixes #21982 . 

**Release Notes** 
Bug fix has been added to the release notes.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by MatthewBowles** [comment]
I get different results in the interface and plot:
image
image

**Review by MatthewBowles — CHANGES_REQUESTED** [review]
The absorption corrections is fine, just fix the inconsistency in the container subtraction plots.

**Comment by brandonhewer** [comment]
@MatthewBowles The issue in the Container Subtraction interface should now be fixed.

**Review by MatthewBowles — APPROVED** [review]
Yep, works fine now.

File: qt/scientific_interfaces/Indirect/ContainerSubtraction.cpp [hunk 1]
```diff
@@ -521,5 +529,19 @@ IAlgorithm_sptr ContainerSubtraction::convertToHistogramAlgorithm(
...
+    MatrixWorkspace_sptr workspace, const std::string &name,
+    const std::string &type, const std::string &value) const {
+  IAlgorithm_sptr shiftLog =
+      AlgorithmManager::Instance().create("AddSampleLog");
+  shiftLog->initialize();
+  shiftLog->setProperty("Workspace", workspace);
+  shiftLog->setProperty("LogName", name);
+  shiftLog->setProperty("LogType", type);
+  shiftLog->setProperty(
+      "LogText", boost::lexical_cast<std::string>(m_uiForm.spShift->value()));
```
Comment (NickDraper) [inline qt/scientific_interfaces/Indirect/ContainerSubtraction.cpp]: You are passing in a value into the function, but are not using it and setting the value from the ui form, why?

**Review by NickDraper — CHANGES_REQUESTED** [review]

File: qt/scientific_interfaces/Indirect/ContainerSubtraction.cpp (hunk 1 above)
Comment (NickDraper) [reply qt/scientific_interfaces/Indirect/ContainerSubtraction.cpp]: This is also the cause of the compiler warning

**Comment by brandonhewer** [comment]
@NickDraper This was a mistake, the value passed into the function should have been used here. I have now updated this.

**Review by NickDraper — APPROVED** [review]

**Pull request merged**